src/utils.jl

```diff
@@ -44,19 +44,83 @@ vcat_arrays_vector(arrays_vector::AbstractVector{Param}) = vcat(Knet.cat1d.(arra
 """
     reset_minibatch_train!(nlp::AbstractKnetNLPModel)
 
-Select a new training minibatch for `nlp`.
-Typically used before a new evaluation of the loss function/gradient.
+Select the first training minibatch for `nlp`.
 """
-reset_minibatch_train!(nlp::AbstractKnetNLPModel) =
-  nlp.current_training_minibatch = rand(nlp.training_minibatch_iterator)
+function reset_minibatch_train!(nlp::AbstractKnetNLPModel)
+  nlp.current_training_minibatch = first(nlp.training_minibatch_iterator)
+  nlp.i_train = 1
```

Comment:
Here, we see that the index 1 is not useful.

Reply:
we use the index 1 as a flag for when data is visited or new iteration has started